internal/controller/postgrescluster/cluster.go

```diff
@@ -53,6 +53,31 @@ func (r *Reconciler) reconcileClusterConfigMap(
 	return clusterConfigMap, err
 }
 
+// patroniLogSize attempts to parse the defined log file storage limit, if configured.
+// If a value is set, this enables volume based log storage and triggers the
+// relevant Patroni configuration. If the value given is less than 25M, the log
+// file size storage limit defaults to 25M and an event is triggered.
+func (r *Reconciler) patroniLogSize(cluster *v1beta1.PostgresCluster) int64 {
+
+	if cluster.Spec.Patroni != nil {
+		if cluster.Spec.Patroni.Logging != nil {
+			if cluster.Spec.Patroni.Logging.StorageLimit != nil {
+
+				sizeInBytes := cluster.Spec.Patroni.Logging.StorageLimit.Value()
+
+				if sizeInBytes < 25000000 {
+					r.Recorder.Eventf(cluster, corev1.EventTypeNormal, "PatroniLogStorageLimitTooSmall",
+						"Configured Patroni log storage limit is too small. File size will default to 25M.")
```

Comment:
⛏️ IIUC, we can use CEL validation for this [eventually](https://kubernetes.io/docs/reference/using-api/cel/) (quantity library). Please add a `TODO(validation)` or so.